### PR TITLE
[NoQA] Fix Virtualised List Error

### PR DESCRIPTION
**Pull request by allgandalf**
### Details

We cannot use `ScrollView` with `SelectionList` because it is  `anti-pattern` and will lead to performance issues and bugs.

With this PR, we are removing the usage of `ScrollView` and adding the extra components as  either `listHeaderContent` or `listFooterContent`
### Fixed Issues

$ https://github.com/Expensify/App/issues/48974
PROPOSAL:

### Tests

* Only on `Dev` environment

PREREQUISITES: all betas or workspace feed beta enables

1. Enable Expensify Card in More Features
2. Add a new bank account and choose this bank for Expensify Card
3. Right after adding a bank and enabling the card on the Expensify Card page go to Settings
4. Go to Settlement Account settings page

Verify that we do not get the error: `VirtualizedLists should never be nested inside plain ScrollViews with the same orientation because it can break windowing and other functionality - use another VirtualizedList-backed container instead.`

---
* Precondition:

1. Comment [this](https://github.com/Expensify/App/blob/2190f6279041ed8260752294d095bbdda76faebe/src/pages/settings/Wallet/TransferBalancePage.tsx#L157) line of code to shouldShow={false}.
2. Have 2 bank accounts set up on wallet page

1. Go to settings/wallet/choose-transfer-account

Verify that we do not get the error: `VirtualizedLists should never be nested inside plain ScrollViews with the same orientation because it can break windowing and other functionality - use another VirtualizedList-backed container instead.`

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A
### QA Steps

N/A
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/82852201-9c29-4547-abcd-b44e4461dd26

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

**Comment by DylanDylann** [comment]
Let's wrap it by View Component. This is draft implementation. YOu can apply more style to View to display the correct UI

```
             <View style={[styles.flexRow]}>
                <SelectionList
                    ......
                />
            </View>
            <MenuItem
                onPress={navigateToAddPaymentMethodPage}
               .......
            />   

```

**Comment by fabioh8010** [comment]
> Let's wrap it by View Component. This is draft implementation. YOu can apply more style to View to display the correct UI
> 
> ```
>              <View style={[styles.flexRow]}>
>                 <SelectionList
>                     ......
>                 />
>             </View>
>             <MenuItem
>                 onPress={navigateToAddPaymentMethodPage}
>                .......
>             />   
> ```

@DylanDylann Your solution doesn't work well with scrolling or when we have several bank accounts, take a look:

https://github.com/user-attachments/assets/921ef9bc-134c-4581-a0eb-464ee5d7de0f

**Comment by fabioh8010** [comment]
@DylanDylann @allgandalf One easy way to solve this particular case is to use the `listFooterContent` of `SelectionList` component, which allows us to render the button at the bottom of the list:

```tsx
            <SelectionList
                sections={[{data}]}
                ListItem={RadioListItem}
                onSelectRow={(value) => {
                    const accountType = value?.bankAccount?.accountType;
                    const accountData = value?.bankAccount?.accountData;
                    selectAccountAndNavigateBack(accountType, accountData);
                }}
                shouldSingleExecuteRowSelect
                shouldUpdateFocusedIndex
                initiallyFocusedOptionKey={walletTransfer?.selectedAccountID?.toString()}
                listFooterContent={
                    <MenuItem
                        onPress={navigateToAddPaymentMethodPage}
                        title={
                            walletTransfer?.filterPaymentMethodType === CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT
                                ? translate('paymentMethodList.addNewBankAccount')
                                : translate('paymentMethodList.addNewDebitCard')
                        }
                        icon={Expensicons.Plus}
                    />
                }
            />
```


https://github.com/user-attachments/assets/b14afe15-526d-49ec-a12b-b120d7334194

I tested on iOS Native, Safari MWeb and Chrome, seems to work well. We should test on Android too.

The only drawback of this approach is that by supplying the `listFooterContent` prop, `SelectionList` won't render the `Show More` button anymore if the list has more than 500 items, but I think it's unlikely for an user to have so many bank accounts. If this is a problem we should explore the `FlatList` "wrapper" solution then.

**Comment by DylanDylann** [comment]
@fabioh8010 Great, It looks fine

**Comment by fabioh8010** [comment]
Here's the full diff in case you want to test with a lot of bank accounts:

```diff
diff --git a/src/pages/settings/Wallet/ChooseTransferAccountPage.tsx b/src/pages/settings/Wallet/ChooseTransferAccountPage.tsx
index 0bbcf957c31..e3deeeeb711 100644
--- a/src/pages/settings/Wallet/ChooseTransferAccountPage.tsx
+++ b/src/pages/settings/Wallet/ChooseTransferAccountPage.tsx
@@ -8,9 +8,9 @@ import getBankIcon from '@components/Icon/BankIcons';
 import * as Expensicons from '@components/Icon/Expensicons';
 import MenuItem from '@components/MenuItem';
 import ScreenWrapper from '@components/ScreenWrapper';
-import ScrollView from '@components/ScrollView';
 import SelectionList from '@components/SelectionList';
 import RadioListItem from '@components/SelectionList/RadioListItem';
+import type {ListItem} from '@components/SelectionList/types';
 import useLocalize from '@hooks/useLocalize';
 import useThemeStyles from '@hooks/useThemeStyles';
 import {getLastFourDigits} from '@libs/BankAccountUtils';
@@ -20,10 +20,15 @@ import * as PaymentMethods from '@userActions/PaymentMethods';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
-import type {AccountData} from '@src/types/onyx';
+import type {AccountData, BankAccount} from '@src/types/onyx';
 import type {BankName} from '@src/types/onyx/Bank';
 import isLoadingOnyxValue from '@src/types/utils/isLoadingOnyxValue';
 
+type AccountListItem = ListItem & {
+    value?: number;
+    bankAccount: BankAccount;
+};
+
 function ChooseTransferAccountPage() {
     const [walletTransfer, walletTransferResult] = useOnyx(ONYXKEYS.WALLET_TRANSFER);
 
@@ -54,7 +59,7 @@ function ChooseTransferAccountPage() {
     const [bankAccountsList] = useOnyx(ONYXKEYS.BANK_ACCOUNT_LIST);
     const selectedAccountID = walletTransfer?.selectedAccountID;
     const data = useMemo(() => {
-        const options = Object.values(bankAccountsList ?? {}).map((bankAccount) => {
+        const options = Object.values(bankAccountsList ?? {}).map((bankAccount): AccountListItem => {
             const bankName = (bankAccount.accountData?.additionalData?.bankName ?? '') as BankName;
             const bankAccountNumber = bankAccount.accountData?.accountNumber ?? '';
             const bankAccountID = bankAccount.accountData?.bankAccountID ?? bankAccount.methodID;
@@ -78,7 +83,31 @@ function ChooseTransferAccountPage() {
                 bankAccount,
             };
         });
-        return options;
+
+        const testOptions = Array.from({length: 20}).map((v, i): AccountListItem => {
+            const {icon, iconSize, iconStyles} = getBankIcon({styles});
+
+            return {
+                value: i,
+                text: `Bank ${i}`,
+                leftElement: (
+                    <View style={[styles.flexRow, styles.alignItemsCenter, styles.mr3]}>
+                        <Icon
+                            src={icon}
+                            width={iconSize}
+                            height={iconSize}
+                            additionalStyles={iconStyles}
+                        />
+                    </View>
+                ),
+                alternateText: `Bank ${i} text`,
+                keyForList: i.toString(),
+                isSelected: false,
+                bankAccount: {},
+            };
+        });
+
+        return options.concat(testOptions);
     }, [bankAccountsList, selectedAccountID, styles, translate]);
 
     if (isLoadingOnyxValue(walletTransferResult)) {
@@ -91,29 +120,29 @@ function ChooseTransferAccountPage() {
                 title={translate('chooseTransferAccountPage.chooseAccount')}
                 onBackButtonPress={() => Navigation.goBack(ROUTES.SETTINGS_WALLET_TRANSFER_BALANCE)}
             />
-            <ScrollView>
-                <SelectionList
-                    sections={[{data}]}
-                    ListItem={RadioListItem}
-                    onSelectRow={(value) => {
-                        const accountType = value?.bankAccount?.accountType;
-                        const accountData = value?.bankAccount?.accountData;
-                        selectAccountAndNavigateBack(accountType, accountData);
-                    }}
-                    shouldSingleExecuteRowSelect
-                    shouldUpdateFocusedIndex
-                    initiallyFocusedOptionKey={walletTransfer?.selectedAccountID?.toString()}
-                />
-                <MenuItem
-                    onPress={navigateToAddPaymentMethodPage}
-                    title={
-                        walletTransfer?.filterPaymentMethodType === CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT
-                            ? translate('paymentMethodList.addNewBankAccount')
-                            : translate('paymentMethodList.addNewDebitCard')
-                    }
-                    icon={Expensicons.Plus}
-                />
-            </ScrollView>
+            <SelectionList
+                sections={[{data}]}
+                ListItem={RadioListItem}
+                onSelectRow={(value) => {
+                    const accountType = value?.bankAccount?.accountType;
+                    const accountData = value?.bankAccount?.accountData;
+                    selectAccountAndNavigateBack(accountType, accountData);
+                }}
+                shouldSingleExecuteRowSelect
+                shouldUpdateFocusedIndex
+                initiallyFocusedOptionKey={walletTransfer?.selectedAccountID?.toString()}
+                listFooterContent={
+                    <MenuItem
+                        onPress={navigateToAddPaymentMethodPage}
+                        title={
+                            walletTransfer?.filterPaymentMethodType === CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT
+                                ? translate('paymentMethodList.addNewBankAccount')
+                                : translate('paymentMethodList.addNewDebitCard')
+                        }
+                        icon={Expensicons.Plus}
+                    />
+                }
+            />
         </ScreenWrapper>
     );
 }

```

**Comment by DylanDylann** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by DylanDylann — APPROVED** [review]

File: src/pages/settings/Wallet/ChooseTransferAccountPage.tsx [hunk 1]
```diff
@@ -20,10 +20,15 @@ import * as PaymentMethods from '@userActions/PaymentMethods';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
-import type {AccountData} from '@src/types/onyx';
+import type {AccountData, BankAccount} from '@src/types/onyx';
 import type {BankName} from '@src/types/onyx/Bank';
 import isLoadingOnyxValue from '@src/types/utils/isLoadingOnyxValue';
 
+type AccountListItem = ListItem & {
```
Comment (fabioh8010) [inline src/pages/settings/Wallet/ChooseTransferAccountPage.tsx:27]:
```suggestion
type BankAccountListItem = ListItem & {
```

NAB

**Review by fabioh8010 — APPROVED** [review]

File: contributingGuides/STYLE.md [hunk 2]
```diff
@@ -1105,6 +1108,48 @@ There are several ways to use and declare refs and we prefer the [callback metho
...
+
+## Correct Approach (Using `SelectionList`)
+
+```jsx
+<SelectionList 
+    sections={[{item}]} 
+    ListItem={RadioListItem} 
+    onSelectRow={handleSelect}
+    ListHeaderComponent={<Text>Header Content</Text>}
+    ListFooterComponent={<Button title="Submit" onPress={handleSubmit} />}
```
Comment (fabioh8010) [inline contributingGuides/STYLE.md:1146]:
```suggestion
    listHeaderComponent={<Text>Header Content</Text>}
    listFooterComponent={<Button title="Submit" onPress={handleSubmit} />}
```

File: contributingGuides/STYLE.md [hunk 3]
```diff
@@ -1105,6 +1108,48 @@ There are several ways to use and declare refs and we prefer the [callback metho
...
+
+## Problem
+
+When using `SelectionList` alongside other components (e.g., `Text`, `Button`), wrapping them inside a `ScrollView` can lead to alignment and performance issues. Additionally, using `ScrollView` with nested `FlatList` or `SectionList` causes the error:
+
+> "VirtualizedLists should never be nested inside plain ScrollViews with the same orientation."
+
+## Solution
+
+The correct approach is to replace Avoid using `ScrollView`. You can add props like `ListHeaderComponent` and `ListFooterComponent` to add other components before or after the list while keeping the layout scrollable.
```
Comment (fabioh8010) [inline contributingGuides/STYLE.md:1122]:
```suggestion
The correct approach is avoid using `ScrollView`. You can add props like `listHeaderComponent` and `listFooterComponent` to add other components before or after the list while keeping the layout scrollable.
```

File: src/pages/settings/Wallet/ChooseTransferAccountPage.tsx (hunk 1 above)
Comment (fabioh8010) [reply src/pages/settings/Wallet/ChooseTransferAccountPage.tsx:27]: Do you mind addressing this @allgandalf ?

File: contributingGuides/STYLE.md [hunk 4]
```diff
@@ -1105,6 +1108,48 @@ There are several ways to use and declare refs and we prefer the [callback metho
 
 We love React and learning about all the new features that are regularly being added to the API. However, we try to keep our organization's usage of React limited to the most stable set of features that React offers. We do this mainly for **consistency** and so our engineers don't have to spend extra time trying to figure out how everything is working. That said, if you aren't sure if we have adopted something, please ask us first.
 
+
+# Handling Scroll Issues with Nested Lists in React Native
```
Comment (fabioh8010) [inline contributingGuides/STYLE.md:1112]:
```suggestion
## Handling Scroll Issues with Nested Lists in React Native
```

We should use H2 like the other sections

File: contributingGuides/STYLE.md [hunk 5]
```diff
@@ -1105,6 +1108,48 @@ There are several ways to use and declare refs and we prefer the [callback metho
...
+
+## Problem
```
Comment (fabioh8010) [inline contributingGuides/STYLE.md:1114]:
```suggestion
### Problem
```

File: contributingGuides/STYLE.md [hunk 6]
```diff
@@ -1105,6 +1108,48 @@ There are several ways to use and declare refs and we prefer the [callback metho
...
+
+When using `SelectionList` alongside other components (e.g., `Text`, `Button`), wrapping them inside a `ScrollView` can lead to alignment and performance issues. Additionally, using `ScrollView` with nested `FlatList` or `SectionList` causes the error:
+
+> "VirtualizedLists should never be nested inside plain ScrollViews with the same orientation."
+
+## Solution
```
Comment (fabioh8010) [inline contributingGuides/STYLE.md:1120]:
```suggestion
### Solution
```

File: contributingGuides/STYLE.md [hunk 7]
```diff
@@ -1105,6 +1108,48 @@ There are several ways to use and declare refs and we prefer the [callback metho
...
+
+The correct approach is avoid using `ScrollView`. You can add props like `listHeaderComponent` and `listFooterComponent` to add other components before or after the list while keeping the layout scrollable.
+
+## Wrong Approach (Using `SelectionList`)
```
Comment (fabioh8010) [inline contributingGuides/STYLE.md:1124]:
```suggestion
### Wrong Approach (Using `SelectionList`)
```

File: contributingGuides/STYLE.md [hunk 8]
```diff
@@ -1105,6 +1108,48 @@ There are several ways to use and declare refs and we prefer the [callback metho
...
+    <SelectionList
+        sections={[{data}]}
+        ListItem={RadioListItem}
+        onSelectRow={handleSelect}
+    />
+    <Button title="Submit" onPress={handleSubmit} />
+</ScrollView>
+```
+
+## Correct Approach (Using `SelectionList`)
```
Comment (fabioh8010) [inline contributingGuides/STYLE.md:1138]:
```suggestion
### Correct Approach (Using `SelectionList`)
```

**Comment by fabioh8010** [comment]
LGTM!

**Review by DylanDylann — APPROVED** [review]

File: contributingGuides/STYLE.md [hunk 9]
```diff
@@ -1105,6 +1108,48 @@ There are several ways to use and declare refs and we prefer the [callback metho
 
 We love React and learning about all the new features that are regularly being added to the API. However, we try to keep our organization's usage of React limited to the most stable set of features that React offers. We do this mainly for **consistency** and so our engineers don't have to spend extra time trying to figure out how everything is working. That said, if you aren't sure if we have adopted something, please ask us first.
 
+
+## Handling Scroll Issues with Nested Lists in React Native
+
+### Problem
+
+When using `SelectionList` alongside other components (e.g., `Text`, `Button`), wrapping them inside a `ScrollView` can lead to alignment and performance issues. Additionally, using `ScrollView` with nested `FlatList` or `SectionList` causes the error:
```
Comment (mountiny) [inline contributingGuides/STYLE.md:1116]:
```suggestion
When using `SelectionList` alongside other components (e.g., `Text`, `Button`), wrapping them inside a `ScrollView` can lead to alignment and performance issues. Additionally, using `ScrollView` with nested `FlatList` or `SectionList` causes this error:
```

**Review by mountiny — APPROVED** [review]
Thank you, one NAB so going to merge

**Pull request merged**